### ISSUE TITLE
2.13: no binary_sensors created

### ISSUE THREAD

**Issue by Mariusthvdb** [robmarkcole/Hue-sensors-HASS#224]
with refactored 2.13, no binary_sensors are created at all (remote and device tracker are ok)

this error is in the logs
```
2020-03-10 11:38:31 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/config/custom_components/hue_custom/data_manager.py", line 137, in async_update_from_bridges
    self.sensors[dev_id].async_write_ha_state()
KeyError: 'SML_00:17:88:01:06:45:41:ea-02'
2020-03-10 11:38:42 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/config/custom_components/hue_custom/data_manager.py", line 137, in async_update_from_bridges
    self.sensors[dev_id].async_write_ha_state()
KeyError: 'SML_00:17:88:01:06:45:41:ea-02'
```

ps my setup resides in /hue_custom so hence the deviating path in the error

**Comment by Kugelfang666** [comment]
Couldn’t be that with 2.13 something broke for multi Bridge scenarios? I had all buttons from one bridge comming up whereas the remotes from the second bridge were all unavailable.

**Comment by Mariusthvdb** [comment]
not sure, not in my case probably, I have all on 1 bridge for now, though a second bridge is indeed online, but without any devices yet.

**Comment by azogue** [comment]
Hi @Mariusthvdb,

> with refactored 2.13, no binary_sensors are created at all (remote and device tracker are ok)

Binary sensors work for me (I have a tap, some hue dimmers, and 2 motion sensors), could you provide log details? (please set custom_components.huesensor: debug in logger)

> ps my setup resides in /hue_custom so hence the deviating path in the error

Please stick to the given code, specific paths have consequences in how the HA setup works, and need to be in sync with the `DOMAIN` inside the code. 

> hope this refactor doesn't introduce the same issue core HA suffers, going unavailable all the time.
> 2.12 bringing back everything immediately

Sorry, but since 2.12, **the data source is the one in HA Core**, so it will suffer the same issues then the official integration. 

Merging the data source to share the calls and reduce the total number was the objective of that version. 2.13 (probably a too quick version) just follows that path.

I have a v2 bridge with > 20 lights, 6 remotes and 2 motion sensors, and I don't experience any unavailable issues. As discussed in other threads, maybe you have too many clients hammering the bridge? is v2?

**Comment by Mariusthvdb** [comment]
> Binary sensors work for me (I have a tap, some hue dimmers, and 2 motion sensors), could you provide log details? (please set custom_components.huesensor: debug in logger)

only log for now was the given above, will retry and add the logging to debug.

> specific paths have consequences in how the HA setup works, and need to be in sync with the DOMAIN inside the code.

I have also set the DOMAIN to hue_custom (always felt huesensor was not the correct name for it, since too confusing compared the the core hue sensors... it is the custom integration after all ;-) 
And, shouldn't be the issue since remotes and device_tracker work fine?

btw, I did try it with the default settings as said above, and that took out the remotes and device_tracker too...

> Sorry, but since 2.12, the data source is the one in HA Core, so it will suffer the same issues then the official integration.

this doesn't sound good... main reason using the Custom integration for me has always been its was rock solid, compared too the truly unreliable performance of the core sensors/lights..Haivng tried only core sensors last week (hoping this would solve the issues of unavailability) and seeing all entities go unavailable made me rush back to the custom integration..

my setup is somewhat bigger than yours though (15 motion sensors, some 40 lights, and a few remotes here and there..)

also v2 bridges here, only 1 with devices for now.
